data/data/gcp/main.tf

```diff
@@ -48,3 +48,15 @@ module "network" {
   worker_subnet_cidr = local.worker_subnet_cidr
   network_cidr       = var.machine_cidr
 }
+
+module "dns" {
+  source = "./dns"
+
+  cluster_id           = var.cluster_id
+  public_dns_zone_name = "??"
```

Comment:
@abhinavdahiya do you have a preference for when I should get this value? 
Terraform won't let me find the public zone by domain name, so I need its name. It seems that at some point during tf var generation I'd have to reach out to GCP and get it. Should that be done here?: 
https://github.com/openshift/installer/blob/master/pkg/asset/cluster/tfvars.go#L187

Reply:
I see we're getting it here: https://github.com/openshift/installer/blob/master/pkg/asset/manifests/dns.go#L103
Maybe there should be a common place where both (tfvars and manifests) can access it?

Reply:
it's fine if use data source to get the public zone like we do for AWS

https://github.com/openshift/installer/blob/11aa8211d062648b20842cf5a512aabdd95bb76e/data/data/aws/route53/base.tf#L1-L3

Reply:
The gcp data source only takes name which is not the same as the domain name:
https://www.terraform.io/docs/providers/google/d/dns_managed_zone.html

Reply:
For now I am calling GetZone and passing it as a var:
https://github.com/openshift/installer/pull/1977/files#diff-bfad726be2a317f0c2bee99d4d62e23eR204

Reply:
for now, it's fine.